### PR TITLE
add support for endpoint type 'VPC', remove support for endpoint_type…

### PR DESCRIPTION
**Pull request by sdevriessbp**
… 'VPC_ENDPOINT'. AWS is discontinuing the use of this, see https://docs.aws.amazon.com/transfer/latest/userguide/create-server-in-vpc.html#deprecate-vpc-endpoint

### CODE REVIEW

**Comment by sdevriessbp** [comment]
please create a tag v0.7.0

File: README.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ module "example-transfer" {
       home_directory = "homedir1"
       role_policy    = null
       ssh_pub_keys   = ["key1", "key2"]
+  endpoint_type          = "VPC"
+  vpc_id                 = "vpc-123456"
+  address_allocation_ids = ["eipalloc-12345", "eipalloc-67890"]
+  subnet_ids             = ["subnet-12345", "subnet-67890"]
```
Comment (svanharmelen) [inline README.md]: Please format your code.

File: main.tf [hunk 2]
```diff
@@ -53,10 +53,12 @@ resource "aws_transfer_server" "default" {
...
 
   dynamic "endpoint_details" {
-    for_each = var.vpc_endpoint_id != null ? list(1) : []
+    for_each = var.vpc_id != null ? list(1) : []
 
     content {
-      vpc_endpoint_id = var.vpc_endpoint_id
+      vpc_id                 = var.vpc_id
+      subnet_ids             = var.subnet_ids
+      address_allocation_ids = var.address_allocation_ids
```
Comment (svanharmelen) [inline main.tf]: Please order these alphabetically.

File: variables.tf [hunk 3]
```diff
@@ -29,8 +29,20 @@ variable "users" {
...
+variable "subnet_ids" {
+  type        = list(string)
+  default     = null
+  description = "Optional Subnet IDs"
+}
+
+variable "address_allocation_ids" {
+  type        = list(string)
+  default     = null
+  description = "Optional Public Elastic IPs"
```
Comment (svanharmelen) [inline variables.tf]: Please order the variables alphabetically.

Additionally I wonder if this shouldn't be just one variable that contains an object. Can these variables be used independently? What if I set a `vpc_id` but omit the others? Will it still work in that case?

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (stefanwb) [inline README.md]:
```suggestion
  endpoint_type          = "VPC"
  vpc_id                 = "vpc-123456"
  address_allocation_ids = ["eipalloc-12345", "eipalloc-67890"]
  subnet_ids             = ["subnet-12345", "subnet-67890"]
```
Can you fix the indenting and sort according to convention?

File: variables.tf [hunk 4]
```diff
@@ -29,8 +29,20 @@ variable "users" {
...
   type        = string
   default     = null
-  description = "An optional VPC endpoint ID"
+  description = "An optional VPC ID"
+}
+
+variable "subnet_ids" {
+  type        = list(string)
+  default     = null
+  description = "Optional Subnet IDs"
```
Comment (stefanwb) [inline variables.tf]:
```suggestion
  description = "A list of subnet IDs used for VPC endpoint"
```

File: variables.tf (hunk 3 above)
Comment (sdevriessbp) [reply variables.tf]: if endpoint_type 'VPC' is used, you require a vpc_id, subnet(s) and eip(s). If endpoint_type 'PUBLIC' (the default) is used, you should not have a vpc_id, subnet(s) and eip(s). So I should change the code (if endpoint_type is VPC then enter vpc_id etc)

Comment (stefanwb) [inline variables.tf]:
```suggestion
  description = "A list of Elastic IPs used for VPC endpoint"
```

Comment (svanharmelen) [reply variables.tf]: Let me call you...

File: variables.tf [hunk 5]
```diff
@@ -29,8 +29,20 @@ variable "users" {
   description = "A map with transfer users and configuration details"
 }
 
-variable "vpc_endpoint_id" {
+variable "vpc_id" {
   type        = string
   default     = null
-  description = "An optional VPC endpoint ID"
+  description = "An optional VPC ID"
```
Comment (stefanwb) [inline variables.tf]:
```suggestion
  description = "The ID of the VPC used for VPC endpoint"
```

File: variables.tf [hunk 6]
```diff
@@ -15,6 +16,17 @@ variable "logging_policy" {
...
 
+variable "name" {
+  type        = string
+  description = "A unique name for this transfer server instance"
+}
+
+variable "subnet_ids" {
+  type        = list(string)
+  default     = null
+  description = "Optional Subnet IDs"
```
Comment (stefanwb) [inline variables.tf]:
```suggestion
  description = "A list of subnet IDs used for VPC endpoint"
```

File: variables.tf [hunk 7]
```diff
@@ -1,6 +1,7 @@
-variable "name" {
-  type        = string
-  description = "A unique name for this transfer server instance"
+variable "address_allocation_ids" {
+  type        = list(string)
+  default     = null
+  description = "Optional Public Elastic IPs"
```
Comment (stefanwb) [inline variables.tf]:
```suggestion
  description = "A list of Elastic IPs used for VPC endpoint"
```

**Review by stefanwb — CHANGES_REQUESTED** [review]
Could you please update the description of the variable `endpoint_type` to match your change?

**Review by stefanwb — DISMISSED** [review]

**Review by svanharmelen — APPROVED** [review]
Updated things a little, but looks good to me now 👍🏻

**Pull request merged**